### PR TITLE
[ADP-3305] Use local-cluster executable in integration tests

### PR DESCRIPTION
**Pull request by paolino**
- [x] Factor out a `withLocalCluster` function from the local-cluster specs
- [x] Use `withLocalCluster` in integration-tests setup 

ADP-3305

### CODE REVIEW

File: lib/local-cluster/process/Cardano/Wallet/Launch/Cluster/Process.hs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                    <> case mMinSeverity of
+                        Nothing -> []
+                        Just minSeverity ->
+                            [ "--min-severity"
+                            , show minSeverity
+                            ]
+                    <> case walletOptions of
+                        NoWallet -> []
+                        WalletPresence mPortNumber ->
+                            ["--wallet-present"]
```
Comment (HeinrichApfelmus) [inline lib/local-cluster/process/Cardano/Wallet/Launch/Cluster/Process.hs:215]: See other comment.

File: lib/integration/framework/Test/Integration/Framework/Setup.hs [hunk 2]
```diff
@@ -342,67 +338,67 @@ recordPoolGarbageCollectionEvents TestingCtx{..} eventsRef =
...
+        _ <- ContT $ \k -> bracketTracer' tr "withServer" $ k ()
+        ((runMonitorQ, runFaucetQ), ToTextTracer clog) <-
+            withLocalCluster "main-integration-tests" NoWallet
+                defaultEnvVars faucetFunds
+        smashUrl <- ContT $ withSMASH clog (toFilePath . absDirOf $ testDir)
+        (np, uri) <-
+            onClusterStart
+                ctx
+                dbDecorator
+                runMonitorQ
```
Comment (HeinrichApfelmus) [inline lib/integration/framework/Test/Integration/Framework/Setup.hs:357]: I'm wondering — do we want the `--wallet-present` option for the `local-cluster` executable at all? 🤔

I think that it's fair game to have the faucet be part of the `local-cluster`, but I don't see a good conceptual reason why the `local-cluster` needs to start a `cardano-wallet` process as well.

As far as the test setup code here is concerned, we cann `local-cluster` with `NoWallet` anyway, and instead start up a wallet in `onClusterStart`.

**Review by HeinrichApfelmus — APPROVED** [review]
Thanks! 😊 

I'd be happy to accept, but I'm unsure about the `--wallet-present` option for the `local-cluster` —  I don't see a good conceptual reason why we would want that. 🤔 (We want a faucet, which is a kind of wallet, but I don't see a good reason why we would want a full `cardano-wallet`.)

File: lib/local-cluster/process/Cardano/Wallet/Launch/Cluster/Process.hs (hunk 1 above)
Comment (paolino) [reply lib/local-cluster/process/Cardano/Wallet/Launch/Cluster/Process.hs:215]: The only good reason why we want it now is that wallet-e2e are using it

File: lib/integration/framework/Test/Integration/Framework/Setup.hs (hunk 2 above)
Comment (paolino) [reply lib/integration/framework/Test/Integration/Framework/Setup.hs:357]: I am open to fix / remove `wallet-e2e` package

Comment (paolino) [reply lib/integration/framework/Test/Integration/Framework/Setup.hs:357]: Ticket ?

Comment (HeinrichApfelmus) [reply lib/integration/framework/Test/Integration/Framework/Setup.hs:357]: > The only good reason why we want it now is that wallet-e2e are using it
> I am open to fix / remove wallet-e2e package

Ah, I see. Hm, the `wallet-e2e` package may contain interesting ideas, but it currently doesn't test anything interesting. 🤔 As a minimum viable refactoring, I would keep the package around, but change our CI to no longer run the tests from the `wallet-e2e` package, allowing us to remove the command line option `--wallet-present`.

> Ticket ?

Feel free to create a ticket — we can discuss tomorrow if needed.

**Pull request merged**